static/js/domain-registration.js

```diff
@@ -0,0 +1,142 @@
+(function() {
+	"use strict";
+
+    const domainRegistration = {
+        init: () => {
+            const domainRegistrationForm = document.getElementById("domainRegistration");
+
+            if (domainRegistrationForm) {
+                domainRegistrationForm.addEventListener("submit", domainRegistration.events.onSubmit, false);  
+            }
+        },
+        checkIfDomainIsSafeAndAvailable: async (domain)=> {
+            if (!domain) {
+                throw new Error("No domain submitted.");
+            }
+            
+            const requestUrl = `/accounts/profile/subdomain?subdomain=${domain}`;
+
+            const response = await fetch(requestUrl, {
+                method: "get",
+                mode: "same-origin",
+                credentials: "same-origin",
+            });
+
+            // Catch Redirect Failure
+            // If the endpoint returns false on a queried subdomain, it means one of two things: 
+            //      1. The domain is already registered
+            //      2. The domain is on the emails/badwords.txt 
+            // 
+            // If we know their request is going to fail, we stop trying to catch the submission and let it fail. 
+            // This will add a messages cookie with the "Domain Not Available" error and reload the page. 
+            if (response.redirected) {
+                return false;
+            }
+
+            if (!response.ok) {
+                const message = `An error has occured: ${response.status}`;
+                throw new Error(message);
+            }
+
+            const status = await response.json();
+
+            return status.available;
+        }, 
+        canModalBeSubmitted: ()=>{
+            const modalConfirmCheckbox = document.querySelector(".js-modal-domain-registration-confirmation-checkbox");
+            const modalSubmit = document.querySelector(".js-modal-domain-registration-submit");
+
+            if (!modalConfirmCheckbox.checked) {
+                modalSubmit.disabled = true;
+                return false;
+            }
+
+            modalSubmit.disabled = false;
+            return true;
+        },
+        events: {
+            onSubmit: async (e) => {
+                e.preventDefault();
+
+                const requestedDomain = document.querySelector(".js-subdomain-value").value;
+                const domainCanBeRegistered = await domainRegistration.checkIfDomainIsSafeAndAvailable(requestedDomain);
+                
+                if (domainCanBeRegistered) {
+                    domainRegistration.modal.open();
+                } else {
+                    // If the domain cannot be registered, submit the form to init an error message.
+                    console.log("Domain not available. Please try again");
```

Comment:
Remove this?

Reply:
Resolved (and added lint rule to catch future violations!)